src/functions.rs

```diff
@@ -44,14 +44,6 @@ macro_rules! nullcheck {
 }
 
 macro_rules! errcheck {
-    ($func:ident($ctx:ident.as_raw() $(, $($args:expr),* $(,)?)?)) => {{
-        let result = $func($ctx.as_raw()$(, $($args),*)?);
-        if result == 0 {
-            Err($crate::Error::GeosError((stringify!($func), $ctx.get_last_error())))
-        } else {
-            Ok(result)
-        }
-    }};
```

Comment:
Still a macro. :p

Reply:
Oh, you mean that you'd prefer `nullcheck` to be a function rather than a macro ? I agree that it's preferable to avoid macro wherever possible, the reason I used a macro here is because I can infer the context and the function name directly from the macro arguments.

So it's possible to just write:
```rs
errcheck!(GEOSArea_r(ctx.as_raw(), self.as_raw(), &mut n))?;
```
instead of:
```rs
let ret = GEOSArea_r(ctx.as_raw(), self.as_raw(), &mut n);
errcheck("GEOSArea_r", ctx, 0, ret)?;
```
Which would feel a bit bloaty when used on every geos call.

Reply:
Oh I didn't notice that you were stringifying the function name. Then yes, let's keep it as a macro. The simplification is already a nice plus. :+1: